src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -152,6 +153,71 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 
 
 /*
+ * BuildDistinctPlan create an returns plan for distinct. Depending on
+ * availability of hash function it chooses HashAgg over Sort/Unique
+ * plans.
+ * This function has a potential performance issue since we blindly set
+ * Plan nodes without looking at cost. We might need to revisit this
+ * if we have performance issues with select distinct queries.
+ */
+static Plan *
+BuildDistinctPlan(Query *masterQuery, Plan *topLevelPlan)
+{
+	bool distinctClausesHashable = true;
+	List *distinctClauseList = masterQuery->distinctClause;
+	Plan *plan = topLevelPlan;
+	List *aggregateTargetList = copyObject(masterQuery->targetList);
+	List *aggregateColumnList = pull_var_clause_default((Node *) aggregateTargetList);
+	ListCell *columnCell = NULL;
+
+	Assert(masterQuery->distinctClause);
+	Assert(!masterQuery->hasDistinctOn);
+
+	/*
+	 * make_agg function requires that columns in aggregate target list
+	 * has valid varno which is OUTER_VAR in our case. We should
```

Comment:
I couldn't see this from `make_agg`. How did you make this inference?

Reply:
Could this be the reason?
https://github.com/citusdata/citus/blob/e202c51fec5408f4539e4eecb76327285654e508/src/backend/distributed/planner/multi_master_planner.c#L107

Reply:
Any update on this? Could you update the comment?